pkg/registries/registries.go

```diff
@@ -14,6 +14,12 @@ func scopeMatchesRegistry(scope, reg string) bool {
 	if reg == scope {
 		return true
 	}
+	if isValidWildcard(reg) {
+		arr := strings.SplitN(reg, ".", 2)
+		if len(arr) == 2 {
+			return strings.HasSuffix(scope, arr[1])
```

Comment:
This should wait until https://github.com/containers/image/pull/1191 (and a possible follow-up) settles the matching semantics, e.g. https://github.com/containers/image/pull/1191#pullrequestreview-631869416 .

Reply:
Actually, rereading that comment, it’s clear that `*.example.com` should match `foo.example.com:5000` and `foo.example.com:5000/bar`, the unclear part is how it would be rewritten if Location/Mirrors are involved.

That doesn’t matter for this code, because ICSP prohibits wildcards. The subscope logic needed here will need to account for wildcard vs. port numbers, but that would probably be a rather different implementation to the existing code anyway.